test/testdrive/pr-24663-regression.td

```diff
@@ -0,0 +1,155 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
```

Comment:
OOC does this test fail without the changes on this PR?

Reply:
No, but it exercises some paths that were hard to get right in this pr!